### DISCUSSION TITLE
Reverts to initialData when key is reset to null

### DISCUSSION THREAD

**Discussion by GriffinSauce** [vercel/swr#1250] [Q&A]
When passing initialData and using conditional fetching the data will reset to initialData when the condition is set to falsy instead of leaving it on the last-fetched data.

Example:
```.js
const { data } = useSWR(
  shouldFetch ? `/api/things` : null,
  apiFetch,
  {
    initialData: dataFromSSR
  },
);
```
1. Change `shouldFetch` to true -> will fetch fresh data
2. Change `shouldFetch` to false -> data reverts to initialData instead of staying "fresh"
---
I've created a workaround that displays the behaviour I expected:
```.js
// SWR hack: don't revalidate when closed but do pass latest data to initialData
const initialDataRef = useRef(dataFromSSR);
const { data } = useSWR(
  shouldFetch ? `/api/things` : null,
  apiFetch,
  {
    initialData: initialDataRef.current,
  },
);
useEffect(
  () => {
    initialDataRef.current = data;
  },
  [data],
);
```
1. Change `shouldFetch` to true -> will fetch fresh data
2. Change `shouldFetch` to false -> last fetched data will be returned

---
I appreciate that resetting data to `null` when no `initialData` is passed makes sense, so I'm not sure what the best solution is but the current behaviour is a bit whacky with this combo of settings.

**Comment by nandorojo** [comment]
Would something like this work? Not sure if `initialData` updates, so maybe this wouldn't work, just an idea.

```js
const { data } = useSWR(
  shouldFetch ? `/api/things` : null,
  apiFetch,
  {
    initialData: shouldFetch ? dataFromSSR : null
  },
)
```
